plans/objchange/lcs.go

```diff
@@ -26,7 +26,12 @@ func LongestCommonSubsequence(xs, ys []cty.Value) []cty.Value {
 
 	for y := 0; y < len(ys); y++ {
 		for x := 0; x < len(xs); x++ {
-			eqV := xs[x].Equals(ys[y])
+			unmarkedX, xMarks := xs[x].Unmark()
+			unmarkedY, yMarks := ys[y].Unmark()
```

Comment:
Seeing `Unmark` and not `UnmarkDeep` here makes me wonder what would happen if we had sensitive values deep inside a list. Something like:

```go
before := cty.ListVal([]cty.Value{
  cty.MapVal(map[string]cty.Value{"a": cty.StringVal("x").Mark("sensitive")}),
  cty.MapVal(map[string]cty.Value{"b": cty.StringVal("y")}),
})
after := cty.ListVal([]cty.Value{
  cty.MapVal(map[string]cty.Value{"a": cty.StringVal("x").Mark("sensitive")}),
  cty.MapVal(map[string]cty.Value{"b": cty.StringVal("y").Mark("sensitive")}),
})
```

I think this would currently panic, as we'd only remove top-level marks and `Equals` would return a marked `cty.Bool`. Perhaps it's as simple as switching to `UnmarkDeep`? I'm really not sure, though.

Reply:
Excellent question, and I can add some tests to explore.

Reply:
And it was (suspiciously, for now) fine with UnmarkDeep!